src/pages/ProductPage/model/functions/functions.ts

```diff
@@ -0,0 +1,25 @@
+import { VIEWED_PRODUCTS_LIMIT } from '@/shared/constants/constants'
+
+import { TProduct } from '../types/productTypes'
+
+/**
+ * Функция добавления товара в список просмотренных товаров viewedProducts в session storage.
+ * @param product: TProduct - текущий товар
+ */
+export const addToViewedProducts = (product: TProduct): void => {
+  const viewedProductsStr = sessionStorage.getItem('viewedProducts') || '[]'
+  const viewedProducts: TProduct[] = JSON.parse(viewedProductsStr)
```

Comment:
так норм работает? Вроде нужно добавить вот так:
```suggestion
  const viewedProducts: TProduct[] = JSON.parse(viewedProductsStr) as TProduct[] 
```